frame/staking/src/lib.rs

```diff
@@ -1360,6 +1460,9 @@ decl_module! {
 			add_weight(3, 0, 0);
 			// Additional read from `on_finalize`
 			add_weight(1, 0, 0);
+
+			let _weight = <NominatorChillTasks<T>>::mutate(|task_scheduler| task_scheduler.execute(1000));
```

Comment:
The weight for decoding the `Vec` needs to be considered here. Or do you intent to benchmark it with the maximum size of the queue? I won't recommend that because the costs can be substantial for longer queues.

Reply:
There a comment in the code that in the current WIP stage I am assuming that everything that the scheduler does it self is for free. Indeed, this is one of the main components that need to be taken into account.